app/services/keycloak/PermsService.scala

```diff
@@ -0,0 +1,36 @@
+package services.keycloak
+
+import org.keycloak.authorization.client.AuthzClient
+import org.keycloak.authorization.client.util.HttpResponseException
+import play.api.Configuration
+
+import java.util.Collections
+import javax.inject.{Inject, Singleton}
+import scala.jdk.CollectionConverters._
+
+@Singleton
+class PermsService @Inject()(config: Configuration) {
+
+  val keycloakConfig = new org.keycloak.authorization.client.Configuration(
+    config.get[String]("auth.url"),
+    config.get[String]("auth.realm"),
+    config.get[String]("auth.client-id"),
+    Collections.singletonMap("secret",config.get[String]("auth.secret-key")), null
+  )
+  val authzClient: AuthzClient = AuthzClient.create(keycloakConfig);
+
+  def checkPermissions(token: String, files: Array[String]): Set[String] = {
```

Comment:
a detail but I would either rename this function to retrieveAuthorizedFiles, or change the implementation to return in one shot 2 sets (one of authorized files, one of unauthorized files) : 

```

def checkPermission(token: String, files: Set[String]): (Set[String], Set[String]) = {
  // .....
  val permNames = Set("aaa", "ccc", "eee", "fff")
  
  val authorized = files intersect permNames
  val unauthorized = files diff permNames
  
  (authorized, unauthorized)
}

val files = Set("aaa", "bbb", "ccc", "ddd", "eee")
val (ok, nok) = checkPermission("token", files)
```

Reply:
nice that's cleaner indeed